### ISSUE TITLE
Issue when using the footnote feature in version 0.8.0.

### ISSUE THREAD

**Issue by BLumia** [erusev/parsedown-extra#123]
I'm currently using parsedown 1.8.0-beta-1 and parsedown-extra 0.8.0 on PHP 5.4.17.

When using the footnote feature, PHP will show a notice text and tell me 
`Notice: Undefined index: element in Parsedown.php on line 334` . ([Parsedown.php L334](https://github.com/erusev/parsedown/blob/0a842fb5b1a9a56428e5ea527b7bf04dda6ce36d/Parsedown.php#L334))

I'm not sure it's a issue of parsedown or parsedown-extra so I post it here...

How to reproduce the issue:

``` php
<?php 
	require_once('./Parsedown.php');
	require_once('./ParsedownExtra.php');
	
	$ParsedownExtra = new ParsedownExtra();
	echo $ParsedownExtra->text(
		"test footnote marker[^1]. test footnote marker two[^2]. \n".
		"[^1]: test marker text\n".
		"[^2]: test marker text two"
	);
```

It seems when we call `extractElement()` at line L267 and L319, the argument doesn't have the 'element' index, hope this info helps.

**Comment by AlfredoRamos** [comment]
Any news on this? I get the same error @aidantwoods 

```php
    protected function extractElement(array $Component)
    {
        if ( ! isset($Component['element']) and isset($Component['markup']))
        {
            $Component['element'] = array('rawHtml' => $Component['markup']);
        }
 
        return $Component['element'];
    }
```

The following code:
```php
$parsedownExtra->text('Parsedown Extra [^1]'.PHP_EOL.'[^1]: http://parsedown.org/extra/');
```

Returns the following data (Laravel error handler):
```php
array:1 [
	"Component" => array:6 [
		"label" => "1"
		"text" => "http://parsedown.org/extra/"
		"hidden" => true
		"type" => "Footnote"
		"identified" => true
		"continuable" => true
	]
]
```

Neither `element` or the `markup` are present in the array, so the condition in `Parsedown::extractElement()` is never met.

**Comment by aidantwoods** [comment]
This was an oversight on my part, and should be resolved when using Parsedown 1.8.0-beta-4 with ParsedownExtra 0.8.0-beta-1.

> I'm currently using [...] parsedown-extra 0.8.0

I should mention that ParsedownExtra 0.8.0 wasn't released yet, the number was only set like that in master (which is a development branch) because the release is upcoming :)

**Comment by AlfredoRamos** [comment]
That was fast! It's working again using the latest beta.

Thanks your your hard work @aidantwoods
